docs/internals/optimizer.rst

```diff
@@ -1096,6 +1097,48 @@ The step LiteralRematerialiser is not required for correctness. It helps deal wi
 ``function f(x) -> y { revert(y, y} }`` where the literal ``y`` will be replaced by its value ``0``,
 allowing us to rewrite the function.
 
+.. _unused-store-eliminator:
+
+UnusedStoreEliminator
+^^^^^^^^^^^^^^^^^^^^^
+
+Optimizer component that removes ``sstore`` and ``mstore`` statements if conditions are met for said removal.
+In case of an ``sstore``, if all outgoing code paths lead to a revert or to another ``sstore`` for which the optimizer
+can tell that it will overwrite the first store, the statement will be removed. For example, the following code
+
+.. code-block:: yul
+
+    {
+        let c := calldataload(0)
+        sstore(c, 1)
+        if c {
+            sstore(c, 2)
+        }
+        sstore(c, 3)
+    }
+
+will produce the code below after the unused store eliminator step is run
+
+.. code-block:: yul
+
+    {
+        {
+            let c := calldataload(0)
+            let _2 := 1
+            if c { let _3 := 2 }
+            sstore(c, 3)
+        }
+    }
```

Comment:
I don't think the step actually add the extra block. I looks like the effect of some of the initial required steps. Same with the extra assignment. When I run just this step in `yulopti` I get this:
```yul
{
    let c := calldataload(0)
    if c { }
    sstore(c, 3)
}
```

Reply:
I've copied this from `if_overwrite_all_branches.yul` test. I would assume that one only uses the `USE` step; does the same go for your `yulopti` run?

Reply:
That test probably runs multiple steps. `yulopti` lets you run steps one by one. It's a simple utility you can use in your local build. Check out [`yulopti.cpp`](https://github.com/ethereum/solidity/blob/develop/test/tools/yulopti.cpp).

Reply:
Yup, you're right. Thanks for pointing out `yulopti`.